### PR TITLE
contribution guidelines

### PR DESCRIPTION
**Pull request by tjungblu**
Adding guidelines for contribution and development, filing bugs etc.
I took this from OLM and modified, so please also check for potential copy&paste errors :)

### CODE REVIEW

File: CONTRIB.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+This document outlines some conventions on commit message formatting, contact points for developers, and other resources to help get contributions into the project.
+
+## Getting started
+
+- Fork the repository on GitHub
+- See the [developer guide](DEV.md) for build instructions
+
+## Reporting bugs and creating issues
+
+Reporting bugs is one of the best ways to contribute. Before filing a bug report, ensure the bug hasn't already been reported by searching through the project [Issues][issues].
```
Comment (pweil-) [inline CONTRIB.md:14]: might be good to mention "have you searched" in the bug template too

File: CONTRIB.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+### Code Review
+
+Contributing PRs with a reasonable title and description can go a long way with helping ease the burden of the review process.
+
+It can be helpful after submitting a PR to self-review your changes. This allows you to communicate sections that reviewers should spend time combing over, asking for feedback on a particular implementation, or providing justification for a set of changes in-line ahead of time.
+
+When opening PRs that are in a rough draft or WIP state create a draft PR. This can help save reviewer's time by communicating the state of a PR ahead of time. Draft PRs can be a good way to get early feedback from reviewers on the implementation, focusing less on smaller details, and more on the general approach of changes.
+
+Each PR must be labeled with at least one "lgtm" label and at least one "approved" label before it can be merged. Maintainers that have approval permissions are listed in the "approvers" column in the root [OWNERS][owners] file.
```
Comment (pweil-) [inline CONTRIB.md:39]: May want to mention here about PR size.  It's ok to make multiple smaller PRs if the work can be broken up into discreet units.  That can help reduce the review burden.

File: CONTRIB.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+## Contribution flow
+
+This is a rough outline of what a contributor's workflow looks like:
+
+- Identify or create an issue.
+- Create a topic branch from where to base the contribution. This is usually the main branch.
+- Make commits of logical units, or squash your change into a single commit.
+- Push changes in a topic branch to a personal fork of the repository.
+- Submit a pull request to the [repository](https://github.com/openshift/must-gather-clean).
```
Comment (pweil-) [inline CONTRIB.md:26]: Run the tests before submitting a PR

**Review by pweil- — APPROVED** [review]
minor suggestions but overall this looks good.  Some other places we can probably poach for best practices:

[PR Process](https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md)
[Code Conventions](https://github.com/kubernetes/community/blob/master/contributors/guide/coding-conventions.md)

**Comment by pweil-** [comment]
lol, too slow, whoops

**Comment by tjungblu** [comment]
I'll send a follow-up

**Pull request merged**